server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetadata.java

```diff
@@ -190,9 +192,15 @@ public static IndexTemplateMetadata readFrom(StreamInput in) throws IOException
         builder.order(in.readInt());
         builder.patterns(in.readList(StreamInput::readString));
         builder.settings(Settings.readSettingsFromStream(in));
-        int mappingsSize = in.readVInt();
-        for (int i = 0; i < mappingsSize; i++) {
-            builder.putMapping(in.readString(), CompressedXContent.readCompressedString(in));
+        if (in.getVersion().onOrAfter(Version.V_5_2_0)) {
+            builder.putMapping(CompressedXContent.readCompressedString(in));
+        } else {
+            int mappingsSize = in.readVInt();
+            assert mappingsSize <= 1 : "There was always a single `default` mapping";
```

Comment:
just out of curiosity, when can it be < 1? since we always write `out.writeVInt(1)`?

Reply:
I wasn't sure if all older versions always had a mapping. They should have, but even if they don't the code would otherwise work so I allowed it.

Reply:
thx!